rust/src/transformations/make_stable_expr/expr_fill_nan/mod.rs

```diff
@@ -36,8 +36,16 @@ where
     // only enforce row-by-row context if the fill expression is not broadcastable
     let expr_domain = if fill.clone().meta().root_names().len() > 0 {
         input_domain.as_row_by_row()
-    } else {
+    } else if let Expr::Literal(value) = fill.clone() {
+        if !value.is_scalar() {
+            return fallible!(MakeTransformation, "fill expression must be broadcastable");
```

Comment:
You rephrased this in the docs, but "broadcastable" in an internal error message is ok with me, if you want to keep it.